### ISSUE TITLE
prefer-destructuring says "Use destructuring" on "super"

### ISSUE THREAD

**Issue by kei-ito** [eslint/eslint#9625]
**Tell us about your environment**

* **ESLint Version: 4.11.0**
* **Node Version: v9.2.0**
* **npm Version: 5.5.1**

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```js
const config = {
    parserOptions: {
        ecmaVersion: 6,
    },
    rules: {
        'prefer-destructuring': ['warn'],
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// eslint-test.js
const code = `
class Node {
    static get inheritance() {
        return ['node'];
    }
}
class Element extends Node {
    static get inheritance() {
        const inheritance = super.inheritance;
        inheritance.push('element');
        return inheritance;
    }
}
console.log(Element.inheritance);
`;
const {Linter} = require('eslint');
const linter = new Linter();
const config = {
    parserOptions: {
        ecmaVersion: 6,
    },
    rules: {
        'prefer-destructuring': ['warn'],
    },
};
const messages = linter.verify(code, config);
for (const message of messages) {
    console.log(message);
}
```

```bash
node ./eslint-test.js
```

**What did you expect to happen?**
I expect `prefer-destructuring` ignores the `const inheritance = super.inheritance;` statement.
`const {inheritance} = super` throws a SyntaxError.

**What actually happened? Please include the actual, raw output from ESLint.**
```
{ ruleId: 'prefer-destructuring',
  severity: 1,
  message: 'Use object destructuring.',
  line: 9,
  column: 15,
  nodeType: 'VariableDeclarator',
  source: '        const inheritance = super.inheritance;',
  endLine: 9,
  endColumn: 40 }
```

**Comment by not-an-aardvark** [comment]
Thanks for reporting, I can reproduce this issue.

**Comment by igidas** [comment]
Also I get "ESlint: Use array destructing. (prefer-destructing)" error with 'this':
`this.searchType = this.searchOptions[0];`

**Comment by kei-ito** [comment]
@igidas `[this.searchType] = this.searchOptions` is valid syntax and prefer-destructuring's suggestion is right.
Here's an example: https://runkit.com/kei-ito/5a123c1c7bcef10013ec0171
